File: Maximum Area of a Piece of cake after horizontal and vertical cuts.cpp
class Solution {
public:
    void getMaxDiff(vector<int> v, int start, int end, long long &diff, int length){
        for(int i = 1; i < v.size(); i++){
            int tmpStart = v[i - 1];
            int tmpEnd = v[i];
            int tmpDiff = tmpEnd - tmpStart;
            
            if(diff < tmpDiff){
                diff = tmpDiff;
                start = tmpStart;
                end = tmpEnd;
            }
        }
        
        diff = diff < length - v.back() ? length - v.back() : diff;
    }
    
    int maxArea(int h, int w, vector<int>& horizontalCuts, vector<int>& verticalCuts) {
        const int MOD = 1000000007;
        long long answer = 0;
        long long rowDiff = 0;
        long long colDiff = 0;
        
        sort(horizontalCuts.begin(), horizontalCuts.end());
        sort(verticalCuts.begin(), verticalCuts.end());
        
        rowDiff = horizontalCuts[0];
        colDiff = verticalCuts[0];
        
        getMaxDiff(horizontalCuts, 0, horizontalCuts[0], rowDiff, h);
        getMaxDiff(verticalCuts, 0, verticalCuts[0], colDiff, w);
        
        answer = (rowDiff * colDiff) % MOD;
        
        return answer;
    }
};

File: Maximum Area of a Piece of cake after horizontal and vertical cuts.js
/**
 * @param {number} h
 * @param {number} w
 * @param {number[]} horizontalCuts
 * @param {number[]} verticalCuts
 * @return {number}
 */
const getMaxDiff = (arr, start, end, diff, length) => {
    for(let i = 1; i < arr.length; i++){
        const tmpStart = arr[i - 1];
        const tmpEnd = arr[i];
        const tmpDiff = tmpEnd - tmpStart;
        
        diff = diff < tmpDiff ? tmpDiff : diff;
    }
    
    return diff < length - arr[arr.length - 1] ? length - arr[arr.length - 1] : diff;
};

const maxArea = (h, w, horizontalCuts, verticalCuts) => {
    const MOD = BigInt(1000000007);
    let rowDiff = 0;
    let colDiff = 0;
    
    horizontalCuts.sort((a, b) => a - b);
    verticalCuts.sort((a, b) => a - b);
    
    rowDiff = horizontalCuts[0];
    colDiff = verticalCuts[0];
    
    rowDiff = getMaxDiff(horizontalCuts, 0, rowDiff, rowDiff, h);
    colDiff = getMaxDiff(verticalCuts, 0, colDiff, colDiff, w);
    
    return (BigInt(rowDiff) * BigInt(colDiff)) % MOD;
};